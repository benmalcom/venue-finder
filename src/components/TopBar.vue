<template>
    <div class="container-fluid top-bar">
        <div class="row col-md-9 mx-auto top-bar-inner">
            <div class="col-md-6 d-flex align-items-center">
                <h6 class="text-left text-white">VENUE <strong>FINDER</strong></h6>
            </div>
            <div class="offset-md-3 col-md-3 mt-3 d-flex align-items-center">
                <div class="media">
                    <label class="mr-2 text-white"><i class="fa fa-map-marker fa-2x"></i></label>
                    <div class="media-body text-white">
                        <h5 class="text-sm mb-0">Cras purus odio, vestibulum in vulputate.</h5>
                        <p class="text-xs current-location mt-0">Current location</p>
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
	export default {
		name: 'TopBar',
        props: {

        },
        data: function () {
            return {

            }
        }
	}
</script>

<style scoped>

</style>
