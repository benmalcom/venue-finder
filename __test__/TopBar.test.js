import { shallow } from 'vue-test-utils'
import TopBar from '../src/components/TopBar.vue';

describe('Component: TopBar.vue', () => {
	let comp;

	beforeEach(() => {
		comp = shallow(TopBar, { // Create a shallow instance of the component
			propsData: {
				position: null
			}
		});
	});

	it('Has the expected html structure', () => {
		expect(comp.element).toMatchSnapshot();
	});

	it('Expects :position prop to be null', () => {
		expect(comp.props().position).toBeNull();
	});

	it('Expects :position prop to be set with lat, long', () => {
		comp.setProps({ position: {
			latitude: 2.2222, longitude: 3.3333
		}});

		expect(comp.vm.position).toBeDefined();
		expect(comp.vm.position.latitude).toEqual(2.2222);
		expect(comp.vm.position.longitude).toEqual(3.3333);
	});

});
