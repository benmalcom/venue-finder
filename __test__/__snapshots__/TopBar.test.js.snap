// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TopBar.vue Has the expected html structure 1`] = `
<div
  class="container-fluid top-bar"
>
  <div
    class="row col-md-9 mx-auto top-bar-inner"
  >
    <div
      class="col-md-6 d-flex align-items-center"
    >
      <h6
        class="text-left text-white"
      >
        VENUE 
        <strong>
          FINDER
        </strong>
      </h6>
    </div>
     
    <div
      class="offset-md-2 col-md-4 mt-3 d-flex align-items-center"
    >
      <div
        class="media"
      >
        <label
          class="mr-2 text-white"
        >
          <i
            class="fa fa-map-marker fa-2x"
          />
        </label>
         
        <div
          class="media-body text-white"
        >
          <h5
            class="text-sm mb-0"
          >
            Location details not available
          </h5>
           
          <p
            class="text-xs current-location mt-0"
          >
            Current location
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
